tests/test_allreduce_fault_tolerance.py

```diff
@@ -66,16 +66,19 @@ async def _run_allreduce(self, group_info: GroupInfo, min_vector_size: int, **kw
                     **kwargs,
                 )
 
-                with self.register_allreduce_group(group_info.group_id, allreduce):
-                    if modes[group_info.peer_ids.index(self.peer_id)] != AveragingMode.AUX:
-                        async for tensor, update in azip(as_aiter(*local_tensors), allreduce):
-                            # all-reduce is performed asynchronously while iterating
-                            tensor.add_(update, alpha=self._averaging_alpha)
-                        self._state_updated.set()
+                self._running_groups[group_info.group_id].set_result(allreduce)
```

Comment:
note to others: this is to adjust for the changes into averagar.py for creating multiple groups